roles/create_mce_crs/tasks/main.yml

```diff
@@ -0,0 +1,230 @@
+    kind: Infrastructure
+    namespace: "default"
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: infraID
+
+- name: Get clusterID
+  kubernetes.core.k8s_info:
+    name: "version"
+    api: config.openshift.io/v1
+    kind: ClusterVersion
+    namespace: "default"
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+  register: clusterID
+
+- name: Create  Import Cluster (ClusterDeployment)
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: hive.openshift.io/v1
+      kind: ClusterDeployment
+      metadata:
+        name: "{{ day2_add_worker_namespace }}"
+        namespace: "{{ day2_add_worker_namespace }}"
+      spec:
+        baseDomain: "{{ base_dns_domain }}"
+        clusterInstallRef:
+          group: extensions.hive.openshift.io
+          kind: AgentClusterInstall
+          name: "{{ day2_add_worker_namespace }}"
+          version: v1beta1
+        clusterMetadata:
+          adminKubeconfigSecretRef:
+            name: "{{ day2_add_worker_namespace }}-admin-kubeconfig"
+          adminPasswordSecretRef:
+            name: "{{ day2_add_worker_namespace }}-admin-password"
+          clusterID: "{{ clusterID.resources[0].spec.clusterID }}"
+          infraID: "{{ infraID.resources[0].status.infrastructureName }}"
+        clusterName: "{{ cluster_name }}"
+        installed: true
+        platform:
+          agentBareMetal:
+            agentSelector:
+              matchLabels:
+                cluster-name: "{{ cluster_name }}"
+        pullSecretRef:
+          name: pull-secret
+
+- name: Prepare for Import
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: cluster.open-cluster-management.io/v1
+      kind: ManagedCluster
+      metadata:
+        name: "{{ day2_add_worker_namespace }}"
+        labels:
+          cloud: auto-detect
+          vendor: auto-detect
+      spec:
+        hubAcceptsClient: true
+
+- name: Wait 10 seconds for ManagedCluster to be created
+  pause:
+    seconds: 10
+
+- name: Obtain the klusterlet-crd.yaml
+  shell: "oc get secret {{ day2_add_worker_namespace }}-import -n {{ day2_add_worker_namespace }} -o jsonpath={.data.crds\\\\\\.yaml} | base64 --decode > {{kubeconfig_dest_dir}}klusterlet-crd.yml"
+
+- name: Obtain the import.yaml
+  shell: "oc get secret {{ day2_add_worker_namespace }}-import -n {{ day2_add_worker_namespace }} -o jsonpath={.data.import\\\\\\.yaml} | base64 --decode > {{kubeconfig_dest_dir}}import.yml"
+
+- name: Apply the klusterlet-crd.yaml
+  command: "oc apply -f {{kubeconfig_dest_dir}}klusterlet-crd.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+
+- name: Apply the import.yaml
+  command: "oc apply -f {{kubeconfig_dest_dir}}import.yml --kubeconfig {{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"

```

Comment:
This needs to be converted from `oc apply` to something like 

```
- name: warn on validation errors, check for unexpected properties
  kubernetes.core.k8s:
    state: present
    definition: "{{ lookup('template', '/testing/deployment.yml') | from_yaml }}"
    validate:
      fail_on_error: no
      strict: yes
```

Reply:
done